config/jobs/ppc64le-cloud/build-docker/postsubmit-build-docker.yaml

```diff
@@ -0,0 +1,149 @@
+            mkdir -p ~/.ssh
+            cp /etc/ssh-key/ssh-privatekey  ~/.ssh/id_rsa
+            chmod 600 ~/.ssh/id_rsa
+            ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+
+            echo "* Start prow-build-docker *"
+            ./prow-build-docker.sh
+            rc=$?
+            [ $rc != 0 ] && echo "ERROR: prow-build-docker exited with code:$rc"
+            exit $rc
+
+          securityContext:
+            privileged: true
+          volumeMounts:
+          - name: docker-root
+            mountPath: /var/lib/docker
+          - name: ssh-key
+            readOnly: true
+            mountPath: /etc/ssh-key
+          env:
+            - name: S3_SECRET_AUTH
+              valueFrom:
+                secretKeyRef:
+                  name: docker-s3-credentials
+                  key: password
+        volumes:
+        - name: docker-root
+          emptyDir: {}
+        - name: ssh-key
+          secret:
+           secretName: git-ssh-docker-ce-build
+
+    - name: postsubmit-build-test-containerd
+      cluster: k8s-ppc64le-cluster
+      decorate: true
+      decoration_config:
+        timeout: 3h
+      run_if_changed: 'job/postsubmit-build-docker'
+      branches:
+        - prow-job-tracking
+      reporter_config:
+        slack:
+          channel: 'prow-job-notifications'
+          job_states_to_report:
+           - success
+           - failure
+           - error
+          report_template: 'Job {{.Spec.Job}} of type {{.Spec.Type}} ended with state {{.Status.State}}. <!subteam^S02N6DWBX0F> <{{.Status.URL}}|View logs>'
+      spec:
+        nodeSelector:
+          dedicated: runtimesTeam
+        tolerations:
+        - key: dedicated
+          operator: "Equal"
+          value: runtimesTeam
+          effect: "NoSchedule"
+        containers:
+        - image: quay.io/powercloud/docker-ce-build
+          command:
+          - /bin/bash
+          args:
+          - -c
+          - |
+            ##
+            # This job is populated with the files from the 'prow-job-tracking' branch.
+            # We switch to the 'main' branch, as it contains the scripts used by this job.
+            ###
+
+            set -o errexit
+            set -o nounset
+            set -o pipefail
+            set -o xtrace
+
+            #Set this to your dev branch for testing
+            WORK_BRANCH=main
+
+            #Timestamp file from the tracking branch (required to access the proper s3 directory)
+            TIMESTAMP_FILE=postsubmit-build-docker
+
+            echo "* Setup the files from git *"
+
+            echo "* Backup timestamp file from the previous job *"
+            cp job/${TIMESTAMP_FILE} /tmp
+
+            echo "* Switch back to the git branch: ${REPO_OWNER}/${REPO_NAME}/${WORK_BRANCH} *"
+            git fetch https://github.com/${REPO_OWNER}/${REPO_NAME} ${WORK_BRANCH}
+            git branch --force ${WORK_BRANCH} FETCH_HEAD
+            git checkout ${WORK_BRANCH}
```

Comment:
Unfortunately no, see point 4 in my [previous comment ](https://github.com/ppc64le-cloud/test-infra/pull/293/files/283b0b32c0d8dac9c388c2a6ef45b56906ac6246#r834270804) above

Reply:
@alunsin yes as you explained in the previous call, I understand that the job initially has the content populated from `prow-job-tracking` branch and you are switching to the main branch to access the scripts in it.

My question was,
Can we not skip L122 and L123 and directly say `git checkout ${WORK_BRANCH}` for switching to main branch?

Reply:
No, that would not work. doing a git checkout <branch> will fail here, as the git branch is not fetched.
That's because Prow's clonerefs [pod utilities ](https://github.com/kubernetes/test-infra/tree/master/prow/cmd#pod-utilities) does a light/partial git clone as explained below and only the current branch (prow-job-tracking)  has been fetched by git.
Here is how `clonerefs` init container does the git clone ( as per the clonerefs container's log):
```
git init
git config user.name ci-robot
git config user.email ci-robot@k8s.io
git fetch https://github.com/alunsin/docker-ce-build.git  --tags --prune
git fetch https://github.com/alunsin/docker-ce-build.git 5ed27dcf83b3bc1425ec529436234a59f7e93fb7
git branch --force prow-job-tracking 5ed27dcf83b3bc1425ec529436234a59f7e93fb7
git checkout 5ed27dcf83b3bc1425ec529436234a59f7e93fb7
```